src/core/public/http/http_service.ts

```diff
@@ -28,19 +28,92 @@ import {
   tap,
 } from 'rxjs/operators';
 
-import { FatalErrorsSetup } from '../fatal_errors';
+import { merge } from 'lodash';
+import { format } from 'url';
 
-interface Deps {
-  fatalErrors: FatalErrorsSetup;
-}
+import { Deps, HttpFetchOptions, HttpBody } from './types';
+import { abortable } from './abortable';
+import { HttpFetchError } from './http_fetch_error';
+
+const JSON_CONTENT = /^(application\/(json|x-javascript)|text\/(x-)?javascript|x-json)(;.*)?$/;
+const NDJSON_CONTENT = /^(application\/ndjson)(;.*)?$/;
 
 /** @internal */
 export class HttpService {
   private readonly loadingCount$ = new Rx.BehaviorSubject(0);
   private readonly stop$ = new Rx.Subject();
 
-  public setup({ fatalErrors }: Deps) {
+  public setup({ basePath, injectedMetadata, fatalErrors }: Deps) {
```

Comment:
FYI I already know that we'll need this exposed in the start contract as well. You should be able to return the exact same interface.